skymp5-server/cpp/server_guest_lib/MpActor.cpp

```diff
@@ -84,6 +89,114 @@ void MpActor::SendToUser(const void* data, size_t size, bool reliable)
+        { 3, {} },
+      } },
+    { "GEAR",
+      {
+        { 1, { 0x00013913, 0x00013922, 0x00012E4D, 0x0001B3A1, 0x0006F39E,
+               0x000D8D52, 0x00056A9E, 0x00012E46, 0x00013912, 0x000D8D55,
+               0x00012EB6, 0x000209A6, 0x0006ff38, 0x0005B69F, 0x0005b6a1,
+               0x0006FF45, 0x000209A5, 0x000209A5, 0x000C5D12, 0x00013911,
+               0x0003619E, 0x00012E49, 0x0001B3A3, 0x0001B3A4, 0x0010594D,
+               0x000D8D50, 0x00018388, 0x00013921, 0x00013921, 0x0001394B,
+               0x0001be1a, 0x000f1229, 0x000209a5, 0x0006c1d9, 0x0004223C,
+               0x0001BE1B, 0x0001BE1B, 0x000BACD7, 0x0006FF37, 0x00013910,
+               0x00013920, 0x00012E4B, 0x0001B39F, 0x0006F398, 0x000D8D4E,
+               0x00056a9D, 0x0006F39B, 0x0001B3A0, 0x00013914, 0x0005C06C,
+               0x0006c1d9, 0x0006ff43, 0x0006C1D8, 0x0003452E, 0x000D191F,
+               0x0003452F, 0x000C36E9 } },
+      } },
+    { "CONS",
+      {
+        { 1, { 0x0004B0BA, 0x00034CDF, 0x0005076E, 0x0001D4EC } },
+        { 2, { 0x0006AC4A, 0x0001B3BD, 0x00064B2E, 0x00064B2F, 0x00023D77 } },
+        { 3, { 0x0000353C } }, // change to dlc id 0xXX00353c
+        { 4, { 0x0003EADE } },
+        { 5, { 0x002E504 } },
+      } }
+  };
+
+int GenerateRandomNumber(int leftBound, int rightBound)
+{
+  std::uniform_int_distribution<> distr(leftBound, rightBound);
+  return distr(gen);
+}
+
+std::string AcknowledgeType(int weaponChance, int gearChance,
+                            int consumablesChance, int nothingChance)
+{
+
+  int chance = GenerateRandomNumber(1, 100);
+  std::string type = "";
+
+  if (chance <= weaponChance && weaponChance != 0) {
+    type = "WEAP";
+    return type;
+  } else if (chance <= (weaponChance + gearChance) && gearChance != 0) {
+    type = "GEAR";
+    return type;
+  } else if (chance <= (weaponChance + gearChance + consumablesChance) &&
+             consumablesChance != 0) {
+    type = "CONS";
+    return type;
+  } else {
+    type = "NOTH";
+    return type;
+  }
+}
+
+int AcknowledgeTier(int tier1Chance, int tier2Chance, int tier3Chance,
+                    int tier4Chance, int tier5Chance)
+{
+  int chance = GenerateRandomNumber(1, 100);
+
+  if (chance < tier1Chance && tier1Chance != 0) {
+    return 1;
+  } else if (chance <= (tier1Chance + tier2Chance) && tier2Chance != 0) {
+    return 2;
+  } else if (chance <= (tier1Chance + tier2Chance + tier3Chance) &&
+             tier3Chance != 0) {
+    return 3;
+  } else if (chance <=
+               (tier1Chance + tier2Chance + tier3Chance + tier4Chance) &&
+             tier4Chance != 0) {
+    return 4;
+  } else {
+    return 5;
+  }
+}
+
+uint32_t GetRandomItem(std::string type)
+{
+    int tier = AcknowledgeTier(100, 20, 10, 9, 1);
+    int item = GenerateRandomNumber(0, lootTable[type][tier].size() - 1);
+    return lootTable[type][tier][item]; 
+}
+
+uint32_t GetSlotItem(int weaponChance, int gearChance, int consumableChance,
```

Comment:
Add comment:
// There are four "slots" in the design document. On each item eat you receive an item corresponding to each slot. Slots have different chances of receiving item types.